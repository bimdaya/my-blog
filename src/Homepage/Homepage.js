import React from 'react';

import Header from '../components/Header';
import Introduction from '../components/Introduction';
import PostCatalogue from '../components/PostCatalogue';
import Footer from '../components/Footer';

const Homepage = ({ allPosts }) => {
  return (
    <>
      <Header />
      <Introduction />
      <PostCatalogue posts={allPosts} />
      <Footer />
    </>
  );
};

export default Homepage;
