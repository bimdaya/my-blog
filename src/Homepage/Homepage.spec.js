import React from 'react';
import { render } from '@testing-library/react';
import Homepage from './Homepage';

describe('Homepage', () => {
  it('renders', () => {
    const allposts = {
      posts: [
        {
          slug: 'test_slug',
          title: 'test title',
          date: '2020.04.05',
          description: 'test description',
        },
      ],
    };
    const { container } = render(<Homepage allposts={allposts} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
