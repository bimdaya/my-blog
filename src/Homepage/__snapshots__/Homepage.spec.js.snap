// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders 1`] = `
<h1>
  Bim's Blog
</h1>
`;
